build-logic/src/main/java/com/squareup/workflow1/buildsrc/shardConnectedChecks.kt

```diff
@@ -0,0 +1,111 @@
+package com.squareup.workflow1.buildsrc
+
+import com.squareup.workflow1.buildsrc.sharding.ShardMatrixYamlTask.Companion.registerYamlShardsTasks
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+
+/**
+ * Create "shard" tasks which collectively depend upon all Android `connectedCheck` tasks in the
+ * entire project.
+ *
+ * Each shard depends upon the `connectedCheck` tasks of some subset of Android projects.
+ * Projects are assigned to a shard by counting the number of `@Test` annotations within their
+ * `androidTest` directory, then associating those projects to a shard in a round-robin fashion.
+ *
+ * These shards are invoked in CI using a GitHub Actions matrix. If the number of shards changes,
+ * the `connectedCheckShardMatrixYamlUpdate` task can automatically update the workflow file so
+ * that they're all invoked.
+ *
+ * The shard tasks are invoked as:
+ * ```shell
+ * # roughly 1/3 of the tests
+ * ./gradlew connectedCheckShard1
+ * # the second third
+ * ./gradlew connectedCheckShard2
+ * # the last third
+ * ./gradlew connectedCheckShard3
+ * ```
+ *
+ * @param target the root project which gets the shard tasks
+ * @param shardCount the number of shard tasks to create.
+ */
+fun shardConnectedCheckTasks(
+  target: Project,
+  shardCount: Int = 3
```

Comment:
Why provide a default, seems like that just invites mistakes where you forget to pass the matrix value and don't notice.

Reply:
I wavered.  I didn't want the configuration option to be out of place in the root `build.gradle.kts`, but I also didn't want it to be hard-coded.  So I found a third bad option and went with that. ;)

Fixed.  It's now a `private const val`.